src/test/java/GildedRoseTest.java

```diff
@@ -1,10 +1,39 @@
 import static org.junit.jupiter.api.Assertions.*;
 import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
 
 class GildedRoseTest {
     @Test
     void gildedRoseUpdatesEachItemWithTheAppropriateUpdater() {
+        String brieItemName = "Brie";
+        String baguetteItemName = "Baguette";
+        Item brieItem = new Item(brieItemName, 0, 0);
+        Item baguetteItem = new Item(baguetteItemName, 0, 0);
+        Item[] items = new Item[] { brieItem, baguetteItem };
+        HashMap<String, Updateable> updaters = new HashMap<String, Updateable>();
```

Comment:
*thought:* This is a very interesting refactoring that honours the requirements (no modifying the Item class!) and I like it a lot 🙌 

But one thing (if we're thinking OOP design) to think about is your public API for creating an `Updater` - some of the implementation details (e.g. your `HashMap`) could be leaking out into the tests. Or, perhaps - imagine if there's a huge bug with `HashMap` discovered tomorrow, it's causing major memory leaks in the application, and you need to replace the data structure. 

You'd need to go inside a few files to change this. In a classic Java-y OOP world, you might look to encapsulate your implementation inside your own `Updater` class - then `GildedRose` would only need to know the API for the public methods (the behaviour).